src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,59 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_USERS_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private final StorageDao storage = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        checkLoginUser(user.getLogin());
+        checkPasswordUser(user.getPassword());
+        checkAgeUser(user.getAge());
+        if (storage.get(user.getLogin()) != null) {
+            throw new RegistrationException("The user with login "
+                    + storage.get(user.getLogin()).getLogin()
```

Comment:
```suggestion
            throw new RegistrationException("The user with login "
                    + user.getLogin()
```
you already have that login